### PR TITLE
FormatWriter: do not overflow when aligning

### LINKED ISSUE #2449: Reformat doesn't truncate in some case
In some case the lines are not correctly truncated. 

This template is a guideline, not a strict requirement.
- **Version**: 2.8.5 and 3.0.0-RC1
- **Integration**: IntelliJ
- **Configuration**:
```
version = 3.0.0-RC1
align.arrowEnumeratorGenerator = true
align.openParenCallSite = true
align = more
danglingParentheses = true
includeCurlyBraceInSelectChains = false
maxColumn = 80
newlines.alwaysBeforeTopLevelStatements = true
newlines.penalizeSingleSelectMultiArgList = false
optIn.breakChainOnFirstMethodDot = true
optIn.configStyleArguments = true
project.git = true
rewrite.rules = [RedundantBraces, RedundantParens, PreferCurlyFors]
runner.optimizer.forceConfigStyleOnOffset = 160
style = IntelliJ # For pretty alignment.
```

## Steps

Given code like this:
```scala
    class ThisIsMyClass(a: Long, b: Long)
    //                                                                          | Position 80
    val taskid = "abcd" + (new Random).nextLong
    val init = 1234567890123L
    val end = 1234567891123L
    val myClass = new ThisIsMyClass(init, end)
    val thisIsMyListOfClasses: ListBuffer[ThisIsMyClass] = ListBuffer()
    val abcdefghijklmnopqrstuvwxyz = new ThisIsMyClass(init, end)
    val abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrst = new ThisIsMyClass(init, end)
```

When I run scalafmt with IntelliJ reformat code:

## Problem

Scalafmt formats code like this:
```scala
    class ThisIsMyClass(a: Long, b: Long)
    //                                                                          | Position 80
    val taskid                                           = "abcd" + (new Random).nextLong
    val init                                             = 1234567890123L
    val end                                              = 1234567891123L
    val myClass                                          = new ThisIsMyClass(init, end)
    val thisIsMyListOfClasses: ListBuffer[ThisIsMyClass] = ListBuffer()
    val abcdefghijklmnopqrstuvwxyz                       = new ThisIsMyClass(init, end)
    val abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrst =
      new ThisIsMyClass(init, end)
```

## Expectation

I would like the formatted output to look like this:
```scala
    class ThisIsMyClass(a: Long, b: Long)
    //                                                                          | Position 80
    val taskid  = "abcd" + (new Random).nextLong
    val init    = 1234567890123L
    val end     = 1234567891123L
    val myClass = new ThisIsMyClass(init, end)
    val thisIsMyListOfClasses: ListBuffer[ThisIsMyClass] =
      ListBuffer()
    val abcdefghijklmnopqrstuvwxyz =
      new ThisIsMyClass(init, end)
    val abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrst =
      new ThisIsMyClass(init, end)
```

## Workaround

I've found that by...

## Notes

I'm not sure if the expectation is correct but I have expected all of the lines with more than 80 characters would be trunctaed

### PR DESCRIPTION
**Pull request by kitbellew**
Fixes #2449. Also improves alignment by detecting longer blocks of lines.

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -626,6 +626,17 @@ for {
 } yield ()
 ```
 
+### `align.allowOverflow`
+
+If this flag is set, as long as unaligned lines did not overflow, we will not check
+whether alignment causes any lines to overflow.
```
Comment (olafurpg) [inline docs/configuration.md:632]:
```suggestion
whether alignment causes any lines to overflow the maxColumn setting. 
```

**Review by olafurpg — APPROVED** [review]
LGTM. It might be worth considering switching to default to allow overflows. I don’t have a strong opinion either way but it appears people who enable alignment seem to appreciate alignment above anything else.

File: docs/configuration.md (hunk 1 above)
Comment (kitbellew) [reply docs/configuration.md:632]: will do

**Comment by kitbellew** [comment]
> LGTM. It might be worth considering switching to default to allow overflows. I don’t have a strong opinion either way but it appears people who enable alignment seem to appreciate alignment above anything else.

options:
* default is true
* false for `some`, true for `more/most`
* false for `some/more`, true for `most`

@olafurpg which one would you choose?

**Pull request merged**